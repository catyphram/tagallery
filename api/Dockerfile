FROM golang:1-alpine

RUN  apk add --update musl-dev gcc git

WORKDIR /go/src/app
COPY . .

RUN go get -d ./...
RUN go build

CMD ["./api"]