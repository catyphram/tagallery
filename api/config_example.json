{
    "database": "tagallery",
    "databaseHost": "localhost:27017",
    "port": 3333,
    "unprocessedImages": "/path/to/new/images"
    
}