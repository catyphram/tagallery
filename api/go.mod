module tagallery.com/api

go 1.15

require (
	github.com/gin-gonic/gin v1.6.3
	github.com/pkg/errors v0.9.1
	go.mongodb.org/mongo-driver v1.4.4
	go.uber.org/zap v1.16.0
)
