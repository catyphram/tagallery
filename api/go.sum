github.com/davecgh/go-spew v1.1.1 h1:vj9j/u1bqnvCEfJOwUhtlOARqs3+rkHYY13jYWTU97c=
github.com/davecgh/go-spew v1.1.1/go.mod h1:J7Y8YcW2NihsgmVo/mv3lAwl/skON4iLHjSsI+c5H38=
github.com/go-chi/chi v3.3.3+incompatible h1:KHkmBEMNkwKuK4FdQL7N2wOeB9jnIx7jR5wsuSBEFI8=
github.com/go-chi/chi v3.3.3+incompatible/go.mod h1:eB3wogJHnLi3x/kFX2A+IbTBlXxmMeXJVKy9tTv1XzQ=
github.com/go-chi/render v1.0.1 h1:4/5tis2cKaNdnv9zFLfXzcquC9HbeZgCnxGnKrltBS8=
github.com/go-chi/render v1.0.1/go.mod h1:pq4Rr7HbnsdaeHagklXub+p6Wd16Af5l9koip1OvJns=
github.com/go-stack/stack v1.8.0 h1:5SgMzNM5HxrEjV0ww2lTmX6E2Izsfxas4+YHWRs3Lsk=
github.com/go-stack/stack v1.8.0/go.mod h1:v0f6uXyyMGvRgIKkXu+yp6POWl0qKG85gN/melR3HDY=
github.com/golang/snappy v0.0.0-20180518054509-2e65f85255db h1:woRePGFeVFfLKN/pOkfl+p/TAqKOfFu+7KPlMVpok/w=
github.com/golang/snappy v0.0.0-20180518054509-2e65f85255db/go.mod h1:/XxbfmMg8lxefKM7IXC3fBNl/7bRcc72aCRzEWrmP2Q=
github.com/google/pprof v0.0.0-20190109223431-e84dfd68c163 h1:beB+Da4k9B1zmgag78k3k1Bx4L/fdWr5FwNa0f8RxmY=
github.com/google/pprof v0.0.0-20190109223431-e84dfd68c163/go.mod h1:zfwlbNMJ+OItoe0UupaVj+oy1omPYYDuagoSzA8v9mc=
github.com/konsorten/go-windows-terminal-sequences v0.0.0-20180402223658-b729f2633dfe/go.mod h1:T0+1ngSBFLxvqU3pZ+m/2kptfBszLMUkC4ZK/EgS/cQ=
github.com/mongodb/mongo-go-driver v0.0.16 h1:iL03HLzhBj4ymNeAT1GdFQe6AUF4dr3h4wKb3xbSRLg=
github.com/mongodb/mongo-go-driver v0.0.16/go.mod h1:NK/HWDIIZkaYsnYa0hmtP443T5ELr0KDecmIioVuuyU=
github.com/pmezard/go-difflib v1.0.0 h1:4DBwDE0NGyQoBHbLQYPwSUPoCMWR5BEzIk/f1lZbAQM=
github.com/pmezard/go-difflib v1.0.0/go.mod h1:iKH77koFhYxTK1pcRnkKkqfTogsbg7gZNVY4sRDYZ/4=
github.com/sirupsen/logrus v1.1.1 h1:VzGj7lhU7KEB9e9gMpAV/v5XT2NVSvLJhJLCWbnkgXg=
github.com/sirupsen/logrus v1.1.1/go.mod h1:zrgwTnHtNr00buQ1vSptGe8m1f/BbgsPukg8qsT7A+A=
github.com/stretchr/testify v1.2.2 h1:bSDNvY7ZPG5RlJ8otE/7V6gMiyenm9RtJ7IUVIAoJ1w=
github.com/stretchr/testify v1.2.2/go.mod h1:a8OnRcib4nhh0OaRAV+Yts87kKdq0PP7pXfy6kDkUVs=
github.com/tkanos/gonfig v0.0.0-20180918200125-3950e265aebb h1:4ppOMaX+DCUq7Fn+Ey1dgJzvYS/bgqEoDdDt9IFwbTc=
github.com/tkanos/gonfig v0.0.0-20180918200125-3950e265aebb/go.mod h1:DaZPBuToMc2eezA9R9nDAnmS2RMwL7yEa5YD36ESQdI=
github.com/xdg/scram v0.0.0-20180814205039-7eeb5667e42c h1:u40Z8hqBAAQyv+vATcGgV0YCnDjqSL7/q/JyPhhJSPk=
github.com/xdg/scram v0.0.0-20180814205039-7eeb5667e42c/go.mod h1:lB8K/P019DLNhemzwFU4jHLhdvlE6uDZjXFejJXr49I=
github.com/xdg/stringprep v1.0.0 h1:d9X0esnoa3dFsV0FG35rAT0RIhYFlPq7MiP+DW89La0=
github.com/xdg/stringprep v1.0.0/go.mod h1:Jhud4/sHMO4oL310DaZAKk9ZaJ08SJfe+sJh0HrGL1Y=
golang.org/x/arch v0.0.0-20181203225421-5a4828bb7045 h1:Pn8fQdvx+z1avAi7fdM2kRYWQNxGlavNDSyzrQg2SsU=
golang.org/x/arch v0.0.0-20181203225421-5a4828bb7045/go.mod h1:cYlCBUl1MsqxdiKgmc4uh7TxZfWSFLOGSRR090WDxt8=
golang.org/x/crypto v0.0.0-20180904163835-0709b304e793 h1:u+LnwYTOOW7Ukr/fppxEb1Nwz0AtPflrblfvUudpo+I=
golang.org/x/crypto v0.0.0-20180904163835-0709b304e793/go.mod h1:6SG95UA2DQfeDnfUPMdvaQW0Q7yPrPDi9nlGo2tz2b4=
golang.org/x/net v0.0.0-20181017193950-04a2e542c03f h1:4pRM7zYwpBjCnfA1jRmhItLxYJkaEnsmuAcRtA347DA=
golang.org/x/net v0.0.0-20181017193950-04a2e542c03f/go.mod h1:mL1N/T3taQHkDXs73rZJwtUhF3w3ftmwwsq0BUmARs4=
golang.org/x/sync v0.0.0-20180314180146-1d60e4601c6f h1:wMNYb4v58l5UBM7MYRLPG6ZhfOqbKu7X5eyFl8ZhKvA=
golang.org/x/sync v0.0.0-20180314180146-1d60e4601c6f/go.mod h1:RxMgew5VJxzue5/jJTE5uejpjVlOe/izrB70Jof72aM=
golang.org/x/sys v0.0.0-20180905080454-ebe1bf3edb33 h1:I6FyU15t786LL7oL/hn43zqTuEGr4PN7F4XJ1p4E3Y8=
golang.org/x/sys v0.0.0-20180905080454-ebe1bf3edb33/go.mod h1:STP8DvDyc/dI5b8T5hshtkjS+E42TnysNCUPdjciGhY=
golang.org/x/text v0.3.0 h1:g61tztE5qeGQ89tm6NTjjM9VPIm088od1l6aSorWRWg=
golang.org/x/text v0.3.0/go.mod h1:NqM8EUOU14njkJ3fqMW+pc6Ldnwhi/IjpwHt7yyuwOQ=
