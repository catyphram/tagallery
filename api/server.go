// e. g. curl http://localhost:3333/images?categories=cat2&categories=cat1&status=unprocessed&last=abc123.jpg

package main

import (
	"fmt"
	"net/http"

	log "github.com/sirupsen/logrus"
	"tagallery.com/api/config"
	"tagallery.com/api/mongodb"
	"tagallery.com/api/router"
)

func main() {
	err := config.Load()
	if err != nil {
		log.WithFields(log.Fields{
			"error": err,
		}).Fatal("An error while loading the configurations has occured")
	}
	_, _, err = mongodb.Init(mongodb.DatabaseOptions{
		Database: config.GetConfig().Database,
		Host:     config.GetConfig().Database_Host,
	})
	if err != nil {
		log.WithFields(log.Fields{
			"error": err,
		}).Fatal("An error during the startup of the mongodb client occured")
	}
	http.ListenAndServe(fmt.Sprintf(":%v", config.GetConfig().Port), router.CreateRouter())
}
