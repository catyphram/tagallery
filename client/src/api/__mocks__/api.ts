import { Category, Image } from '@/models';

export const loadCategories = jest.fn(async (): Promise<Category[]> => {
  return await [
    { name: 'Category 1', key: 'cat1', description: 'Category 1' },
    { name: 'Category 2', key: 'cat2', description: 'Category 2' },
    { name: 'Category 3', key: 'cat3', description: 'Category 3' },
    { name: 'Category 4', key: 'cat4', description: 'Category 4' },
    { name: 'Category 5', key: 'cat5', description: 'Category 5' },
    { name: 'Category 6', key: 'cat6', description: 'Category 6' },
  ];
});

export const loadImages = jest.fn(async (): Promise<Image[]> => {
  return await [
    { file: 'image0.jpg' },
    { file: 'image1.jpg' },
    { file: 'image2.jpg' },
    { file: 'image3.jpg' },
    { file: 'image4.jpg' },
    { file: 'image5.jpg' },
    { file: 'image6.jpg' },
    { file: 'image7.jpg' },
    { file: 'image8.jpg' },
    { file: 'image9.jpg' },
  ] as Image[];
});

export const updateImage = jest.fn(async (image): Promise<boolean> => {
  return await true;
});
