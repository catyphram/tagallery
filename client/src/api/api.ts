import { Category, Image } from '@/models';

/**
 * loadCategories loads the categories from the API.
 */
export const loadCategories = async (): Promise<Category[]> => {
  return await [
      { name: 'Category 1', key: 'cat1', description: 'Category 1' },
      { name: 'Category 2', key: 'cat2', description: 'Category 2' },
  ];
};

// Additional parameter for the local stub only so the URL differs on reload
// and Vue updates the images in the HTML.
let counter = 0;

/**
 * loadImages loads a batch of images from the API.
 * @todo: Add parameters for categories, mode and lastImage filter
 * Also don't forget to cancel the fetch request when another request is sent.
 */
export const loadImages = async (): Promise<Image[]> => {
  const stubImages = [] as Image[];
  for (let i = 0; i < 30; i++) {
    stubImages.push({
      file: `https://picsum.photos/400/400/?random&key=${counter}-${i}`,
    } as Image);
  }

  counter++;

  return await stubImages;
};
