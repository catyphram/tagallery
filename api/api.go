package main

import "tagallery.com/api/server"

func main() {
	server.StartServer()
}
