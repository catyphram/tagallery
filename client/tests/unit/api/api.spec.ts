import { loadCategories } from '@/api/api';

describe('api.ts', () => {
  it('loadCategories should load and return the categories', async () => {
    expect(await loadCategories()).toMatchSnapshot();
  });
});
