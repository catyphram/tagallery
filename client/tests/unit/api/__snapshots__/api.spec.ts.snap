// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`api.ts loadImages should load and return a batch of images 1`] = `
Array [
  Object {
    "file": "https://picsum.photos/400/400/?random&key=0-0",
  },
  Object {
    "file": "https://picsum.photos/400/400/?random&key=0-1",
  },
  Object {
    "file": "https://picsum.photos/400/400/?random&key=0-2",
  },
  Object {
    "file": "https://picsum.photos/400/400/?random&key=0-3",
  },
  Object {
    "file": "https://picsum.photos/400/400/?random&key=0-4",
  },
  Object {
    "file": "https://picsum.photos/400/400/?random&key=0-5",
  },
  Object {
    "file": "https://picsum.photos/400/400/?random&key=0-6",
  },
  Object {
    "file": "https://picsum.photos/400/400/?random&key=0-7",
  },
  Object {
    "file": "https://picsum.photos/400/400/?random&key=0-8",
  },
  Object {
    "file": "https://picsum.photos/400/400/?random&key=0-9",
  },
  Object {
    "file": "https://picsum.photos/400/400/?random&key=0-10",
  },
  Object {
    "file": "https://picsum.photos/400/400/?random&key=0-11",
  },
  Object {
    "file": "https://picsum.photos/400/400/?random&key=0-12",
  },
  Object {
    "file": "https://picsum.photos/400/400/?random&key=0-13",
  },
  Object {
    "file": "https://picsum.photos/400/400/?random&key=0-14",
  },
  Object {
    "file": "https://picsum.photos/400/400/?random&key=0-15",
  },
  Object {
    "file": "https://picsum.photos/400/400/?random&key=0-16",
  },
  Object {
    "file": "https://picsum.photos/400/400/?random&key=0-17",
  },
  Object {
    "file": "https://picsum.photos/400/400/?random&key=0-18",
  },
  Object {
    "file": "https://picsum.photos/400/400/?random&key=0-19",
  },
  Object {
    "file": "https://picsum.photos/400/400/?random&key=0-20",
  },
  Object {
    "file": "https://picsum.photos/400/400/?random&key=0-21",
  },
  Object {
    "file": "https://picsum.photos/400/400/?random&key=0-22",
  },
  Object {
    "file": "https://picsum.photos/400/400/?random&key=0-23",
  },
  Object {
    "file": "https://picsum.photos/400/400/?random&key=0-24",
  },
  Object {
    "file": "https://picsum.photos/400/400/?random&key=0-25",
  },
  Object {
    "file": "https://picsum.photos/400/400/?random&key=0-26",
  },
  Object {
    "file": "https://picsum.photos/400/400/?random&key=0-27",
  },
  Object {
    "file": "https://picsum.photos/400/400/?random&key=0-28",
  },
  Object {
    "file": "https://picsum.photos/400/400/?random&key=0-29",
  },
]
`;
